File: domain/src/main/java/home/oleg/placenearme/repositories/FavoriteVenuesRepository.java
package home.oleg.placenearme.repositories;

import java.util.List;

import home.oleg.placenearme.models.DetailedVenue;
import io.reactivex.Completable;
import io.reactivex.Flowable;

public interface FavoriteVenuesRepository {

    Flowable<List<DetailedVenue>> observeFavorites();

    Completable addToFavorite(DetailedVenue venue);

    Completable deleteFromFavorite(DetailedVenue venue);

}

File: core-domain/src/main/java/home/oleg/placenearme/repositories/FavoriteVenuesRepository.kt
package home.oleg.placenearme.repositories

import home.oleg.placenearme.models.DetailedVenue
import io.reactivex.Completable
import io.reactivex.Flowable

interface FavoriteVenuesRepository {

    fun observeFavorites(): Flowable<List<DetailedVenue>>

    fun addToFavorite(venue: DetailedVenue): Completable

    fun deleteFromFavorite(venue: DetailedVenue): Completable

}
